src/cueUtilities.js

```diff
@@ -196,139 +220,193 @@ module.exports = function (params, cy, api) {
-					&& cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-					if(opts.undoable && !ur)
-						ur = cy.undoRedo({
-							defaultActions: false
-						});
-					if(api.isCollapsible(node))
-						if (opts.undoable){
-							ur.do("collapse", {
-								nodes: node,
-								options: opts
-							});
-						}
-						else
-							api.collapse(node, opts);
-				else if(api.isExpandable(node))
-					if (opts.undoable)
-						ur.do("expand", {
-							nodes: node,
-							options: opts
-						});
-					else
-						api.expand(node, opts);
-					}
-			}
-		});
+        function selectedGroupsContainsGroup(group) {
+          for (let i = 0; i < selectedGroups.length; i++) {
+            if (selectedGroups[i].id() === group.id()) {
+              return true;
+            }
+          }
+          return false;
+        }
+
+        cy.on('mouseout', 'node', data.eMouseOut = function(e) {
+          let node = this;
+          if (mouseIsHoveringOver(node)) {
+            clearDraws();
+            hoveredGroup = null;
+            drawCuesForSelectedGroups();
+          }
+        });
+
+        cy.on('mouseover', 'node', data.eMouseOver = function (e) {
+          let node = this;
+          if (isAGroup(node)) {
+            // clear draws if any
+            if ( hoveredGroup && hoveredGroup.id() != node.id() ) {
+              clearDraws();
+              if (options().appearOnGroupSelect) {
+                drawCuesForSelectedGroups();
+              }
+            }
+            if (!selectedGroupsContainsGroup(node)) {
+              drawExpandCollapseCue(node);
+              hoveredGroup = node;
+            }
+          } else {
+            // needed incase we hover over a regular node inside a group
+            clearDraws();
+            hoveredGroup = null;
+            drawCuesForSelectedGroups();
+          }
+        });
+
+        var oldMousePos = null, currMousePos = null;
+        cy.on('mousedown', 'node', data.eMouseDown = function(e){
+          oldMousePos = e.renderedPosition || e.cyRenderedPosition
+        });
+        cy.on('mouseup', 'node', data.eMouseUp = function(e){
+          currMousePos = e.renderedPosition || e.cyRenderedPosition
+        });
+
+        cy.on('grab', 'node', data.eGrab = function (e) {
+          preventDrawing = true;
+        });
+
+        cy.on('free', 'node', data.eFree = function (e) {
+          preventDrawing = false;
+        });
+
+        cy.on('position', 'node', data.ePosition = function () {
+          if (hoveredGroup) {
+            clearDraws();
+            if (options().appearOnGroupSelect) {
+              drawCuesForSelectedGroups();
+            }
+          }
+        });
+
+        cy.on('remove', 'node', data.eRemove = function () {
+          if (!options().appearOnGroupSelect) {
```

Comment:
Why does removal of nodes only clear the drawing and hovered group if not using appearOnGroupSelect? Would think this would be independent of that feature.

Reply:
For some reason, hovering over a regular node calls this event. We may not even need this at all. mouseout will be called to clearDraw() anyways.